gridscale/resource_gridscale_storage_clone.go

```diff
@@ -0,0 +1,166 @@
+package gridscale
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/gridscale/gsclient-go/v3"
+)
+
+func resourceGridscaleStorageClone() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGridscaleStorageCloneCreate,
+		Read:   resourceGridscaleStorageRead,
+		Delete: resourceGridscaleStorageDelete,
+		Update: resourceGridscaleStorageUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"source_storage_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "ID of a storage instance which will be cloned",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The human-readable name of the object. It supports the full UTF-8 charset, with a maximum of 64 characters.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"capacity": {
+				Type:        schema.TypeInt,
+				Description: "The capacity of a storage in GB.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"location_uuid": {
+				Type:        schema.TypeString,
+				Description: "Helps to identify which datacenter an object belongs to.",
+				Computed:    true,
+			},
+			"storage_type": {
+				Type:        schema.TypeString,
+				Description: "(one of storage, storage_high, storage_insane)",
+				Optional:    true,
+				Computed:    true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					valid := false
+					for _, stype := range storageTypes {
+						if v.(string) == stype {
+							valid = true
+							break
+						}
+					}
+					if !valid {
+						errors = append(errors, fmt.Errorf("%v is not a valid storage type. Valid types are: %v", v.(string), strings.Join(storageTypes, ",")))
+					}
+					return
+				},
+			},
+			"license_product_no": {
+				Type:        schema.TypeInt,
+				Description: "If a template has been used that requires a license key (e.g. Windows Servers) this shows the product_no of the license (see the /prices endpoint for more details).",
```

Comment:
Maybe make `"product_no"` → `"product number"`

Reply:
This would make it inconsistent with the `storage` resource (and the response from gridscale's server).